api/src/main/java/io/minio/credentials/MinioClientConfigProvider.java

```diff
@@ -0,0 +1,117 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.Locale;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Credential provider using MinioClient configuration file. */
+public class MinioClientConfigProvider extends EnvironmentProvider {
+  private final String filename;
+  private final String alias;
+  private final ObjectMapper mapper;
+
+  public MinioClientConfigProvider(@Nullable String filename, @Nullable String alias) {
+    if (filename != null && filename.isEmpty()) {
+      throw new IllegalArgumentException("Filename must not be empty");
+    }
+
+    if (alias != null && alias.isEmpty()) {
+      throw new IllegalArgumentException("Alias must not be empty");
+    }
+
+    this.filename = filename;
+    this.alias = alias;
+    this.mapper = new ObjectMapper();
+    this.mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    this.mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+  }
+
+  /**
+   * Retrieve credentials in provided alias or MINIO_ALIAS or "s3" alias in configuration file from
+   * provided filename or AWS_SHARED_CREDENTIALS_FILE environment variable or file .aws/credentials
+   * in user's home directory.
+   */
+  @Override
+  public Credentials fetch() {
+    String filename = this.filename;
+    if (filename == null) {
+      filename = getProperty("MINIO_SHARED_CREDENTIALS_FILE");
+    }
+    if (filename == null) {
+      String mcDir = ".mc";
+      if (System.getProperty("os.name").toLowerCase(Locale.US).contains("windows")) {
+        mcDir = "mc";
+      }
+
+      filename = Paths.get(System.getProperty("user.home"), mcDir, "config.json").toString();
+    }
+
+    String alias = this.alias;
+    if (alias == null) {
+      alias = getProperty("MINIO_ALIAS");
+    }
+    if (alias == null) {
+      alias = "s3";
+    }
+
+    try (InputStream is = new FileInputStream(filename)) {
+      McConfig config =
+          mapper.readValue(new InputStreamReader(is, StandardCharsets.UTF_8), McConfig.class);
+      Map<String, String> values = config.get(alias);
+      if (values == null) {
+        throw new IllegalStateException(
+            "Alias " + alias + " does not exist in MinioClient configuration file");
+      }
+
+      String accessKey = values.get("accessKey");
+      String secretKey = values.get("secretKey");
+
+      if (accessKey == null) {
+        throw new IllegalStateException(
+            "Access key does not exist in alias " + alias + " in MinioClient configuration file");
+      }
+
+      if (secretKey == null) {
+        throw new IllegalStateException(
+            "Secret key does not exist in alias " + alias + " in MinioClient configuration file");
+      }
+
+      return new Credentials(accessKey, secretKey, null, null);
```

Comment:
you will have to handle `bucket` vs `Path` style access also. In that case, the format has changed recently, so both of them have to be supported

Reply:
@kannappanr 
Can you explain what is  bucket vs Path style access? Also can you provide new format change?

Reply:
with the new format
```
"default": {
                        "url": "http://localhost:9000",
                        "accessKey": "minioadmin",
                        "secretKey": "minioadmin",
                        "api": "s3v4",
                        "path": "auto"
                },
```
Here path has two possible values `auto` and `off`, `off` means it is bucket style access. If the endpoint is AWS, you still have to do bucket style access

Reply:
with old format, 
```
 "s3": {
                        "url": "https://s3.amazonaws.com",
                        "accessKey": "YOUR-ACCESS-KEY-HERE",
                        "secretKey": "YOUR-SECRET-KEY-HERE",
                        "api": "S3v4",
                        "lookup": "dns"
                }

```
`lookup` will have one of three values `dns, path, auto` `dns` means it is bucket style access, in the case of auto, if the endpoint is AWS, we should do bucket style access